pyg_lib/csrc/ops/cuda/matmul_kernel.cu

```diff
@@ -45,6 +74,39 @@ void grouped_matmul_out_kernel(const at::TensorList input,
       cutlass::arch::OpMultiplyAdd                   // Operation
       >::GemmKernel;
 
+  // TODO (matthias) Allow for other types than `float`.
+  // TODO (matthias) Are these attributes correctly set?
+  int grouped_shared_mem = shared_memory_for_kernel<DefaultGemmKernel>();
+  int shared_mem_per_sm = shared_memory_per_sm();
+  if (grouped_shared_mem < shared_mem_per_sm) {
+    using GemmKernel = DefaultGemmKernel;
```

Comment:
Move the definition of `DefaultGemmKernel` to here?

Reply:
Since a lot of args are the same, I suggest we define the key variables separately and set them according to `shared_mem`, then we use a single `using` with those variables. Otherwise it is hard to read the differences of two modes.